src/id3v2.cpp

```diff
@@ -3041,10 +3041,10 @@ void APar_ID3FrameAmmend(AtomicInfo *id32_atom,
     } else if (targetFrame->ID3v2_Frame_ID == ID3v2_FRAME_CONTENTTYPE) {
       uint8_t genre_idx = ID3StringGenreToInt(frame_payload);
       if (genre_idx != 0xFF) {
-        char genre_str_idx[2];
+        char genre_str_idx[3];
         genre_str_idx[0] = 0;
         genre_str_idx[1] = 0;
-        genre_str_idx[1] = 0;
+        genre_str_idx[2] = 0;
         sprintf(genre_str_idx, "%u", genre_idx);
```

Comment:
Urgh, nasty!
However, this should probably use a 4 byte buffer so that there is room for the maximum unsigned decimal u8 number plus its null terminator.

```
  char genre_str_idx[4];
  snprintf(genre_str_idx, sizeof(genre_str_idx), "%u", genre_idx);
```